### ISSUE TITLE
Page Kept Loading when trying to Mask RCNN

### ISSUE THREAD

**Issue by NicksonYap** [jsbroks/coco-annotator#204]
Hi,

I've managed to import models for DEXTR and Mask RCNN

DEXTR by itself, runs nicely

However, when configuring Mask RCNN in .yml,

![image](https://user-images.githubusercontent.com/5380731/57193756-5801a080-6f71-11e9-9ed6-999371a9029e.png)

![image](https://user-images.githubusercontent.com/5380731/57193803-eaa23f80-6f71-11e9-8299-d5e87d5c1ece.png)

the page kept loading

![image](https://user-images.githubusercontent.com/5380731/57193745-3bfdff00-6f71-11e9-9956-c7a9402d5d7f.png)

and i kept getting `[CRITICAL] WORKER TIMEOUT ` in console:

```
 2019-05-05 12:03:12.486775: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
annotator_webserver | 2019-05-05 12:03:12.491671: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3482350000 Hz
annotator_webserver | 2019-05-05 12:03:12.492722: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x564d12821740 executing computations on platform Host. Devices:
annotator_webserver | 2019-05-05 12:03:12.492775: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
annotator_webserver | [2019-05-05 12:03:14 +0000] [11] [INFO] Loaded MaskRCNN model: /models/mask_rcnn_coco.h5
annotator_webserver | [2019-05-05 12:03:30 +0000] [7] [CRITICAL] WORKER TIMEOUT (pid:11)
annotator_webserver | [2019-05-05 12:03:30 +0000] [11] [INFO] Worker exiting (pid: 11)
annotator_webserver | Building network based on ResNet 101 and PSP module expecting inputs of shape (512, 512) predicting 1 classes
annotator_webserver | PSP module will interpolate to a final feature map size of (64, 64)
annotator_webserver | [2019-05-05 12:03:31 +0000] [112] [INFO] Booting worker with pid: 112
annotator_webserver | Using TensorFlow backend.
annotator_webserver | WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
annotator_webserver | Instructions for updating:
annotator_webserver | Colocations handled automatically by placer.
annotator_webserver | WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/mask_rcnn-2.1-py3.6.egg/mrcnn/model.py:772: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
annotator_webserver | Instructions for updating:
annotator_webserver | Use tf.cast instead.
annotator_webserver | 2019-05-05 12:03:41.628291: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
annotator_webserver | 2019-05-05 12:03:41.633320: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3482350000 Hz
annotator_webserver | 2019-05-05 12:03:41.634828: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x564d127ff9a0 executing computations on platform Host. Devices:
annotator_webserver | 2019-05-05 12:03:41.634880: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
annotator_webserver | [2019-05-05 12:03:44 +0000] [112] [INFO] Loaded MaskRCNN model: /models/mask_rcnn_coco.h5
annotator_webserver | [2019-05-05 12:04:01 +0000] [7] [CRITICAL] WORKER TIMEOUT (pid:112)
annotator_webserver | [2019-05-05 12:04:01 +0000] [112] [INFO] Worker exiting (pid: 112)
annotator_webserver | Building network based on ResNet 101 and PSP module expecting inputs of shape (512, 512) predicting 1 classes
annotator_webserver | PSP module will interpolate to a final feature map size of (64, 64)
annotator_webserver | [2019-05-05 12:04:02 +0000] [211] [INFO] Booting worker with pid: 211
annotator_webserver | Using TensorFlow backend.
annotator_webserver | WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
annotator_webserver | Instructions for updating:
annotator_webserver | Colocations handled automatically by placer.
annotator_webserver | WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/mask_rcnn-2.1-py3.6.egg/mrcnn/model.py:772: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
annotator_webserver | Instructions for updating:
annotator_webserver | Use tf.cast instead.
annotator_webserver | 2019-05-05 12:04:13.396334: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
annotator_webserver | 2019-05-05 12:04:13.401586: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 3482350000 Hz
annotator_webserver | 2019-05-05 12:04:13.403752: I tensorflow/compiler/xla/service/service.cc:150] XLA service 0x564d12800240 executing computations on platform Host. Devices:
annotator_webserver | 2019-05-05 12:04:13.403797: I tensorflow/compiler/xla/service/service.cc:158]   StreamExecutor device (0): <undefined>, <undefined>
annotator_webserver | [2019-05-05 12:04:15 +0000] [211] [INFO] Loaded MaskRCNN model: /models/mask_rcnn_coco.h5
```

The same goes when i add my own model and classes:

```
 - MASK_RCNN_FILE=/models/mask_rcnn_zuppa_0149.h5
 - MASK_RCNN_CLASSES=BG,Sour,Tiger,Flower,Lychee,Milo
```

I'm in commit: 303c2e8403b235e686913f060d9ef549484d9668

Regards,

Nickson

**Comment by jsbroks** [comment]
Take a look at #199. You'll want to increase gunicon timout

**Comment by NicksonYap** [comment]
@jsbroks yes, it solved the issue and loaded 👍 

However, I don't know what tool to use to run Mask RCNN on an image, any tips?

**Comment by NicksonYap** [comment]
I checked the source and it seems there is no GUI implemented yet, but there's an API
How did you imagine the interface to be?
I may be able to help on that 😄

**Comment by jsbroks** [comment]
@NicksonYap Please look at #176

**Comment by jsbroks** [comment]
Page loading will be fixed on completion of #153
